quiz.js

```diff
@@ -0,0 +1,459 @@
+    scoreOutput += "<p> Your score is " + score + " out of " + totalQuestions + ". " + message + "</p>";
+
+    return scoreOutput;
+
+}
+
+function generateIncorrectQuestions(incorrectQuestions) {
+
+    var output = "";
+    output += generateIncorrectSummary(incorrectQuestions);
+    output += generateIncorrectDetail(incorrectQuestions);
+
+    return output;
+}
+
+// list the incorrect answers
+function generateIncorrectSummary(incorrectQuestions) {
+    var output = "<p>The following questions were incorrect:</p><ul>";
+    // list the numbers of the incorrect questions
+    for (var y = 0; y < incorrectQuestions.length; y++) {
+        output += "<li class='question incorrect'>" + incorrectQuestions[y].question.number + "</li>";
+
+    }
+    output += "</ul>";
+    output += "</div>";
+    
+    return output;
+
+}
+
+// list the incorrect answers in full
+function generateIncorrectDetail(incorrectQuestions) {
+    
+    var output = "";
+    for (x = 0; x < incorrectQuestions.length; x++) {
+
+        output += "<div class='box'>";
+        output += "<legend class='question-heading'>" + incorrectQuestions[x].question.number + ". " + incorrectQuestions[x].question.title + "</legend>";
+
+        for (var j = 0; j < incorrectQuestions[x].question.answers.length; j++) {
+            output += "<div class='questions'>";
+            output += "<li class='question ";
+            if (j == incorrectQuestions[x].question.correctAnswer) {
+                output += "correct'>";
+            } else if (j == incorrectQuestions[x].incorrectIndex) {
+                output += "incorrect'>";
+            } else {
+                output += "other'>";
+            }
+            output += "<label>" + incorrectQuestions[x].question.answers[j] + "</label>";
+            output += "</li>";
+            output += "</div>";
+        }
+        output += "</div>";
+
+    }
+    
+    return output;
+
+}
+
+/* ========
+
+Calculate Score
+
+=========== */
+
+function calcScore() {
+
+    // store the incorrect questions as objects in an array
+    var incorrectQuestions = [];
+
+    // set the user's score
+    var score = 0;
+
+    // a is the current question
+    for (var a = 0; a < questions.length; a++) {
+
+        // store correct answer for certain question 
+        var correct = questions[a].correctAnswer;
+
+        // get the radio buttons of current question to loop through and see if they're checked
+        var questionAnswers = document.querySelectorAll('[name=q' + questions[a].number + ']');
+
+        // loop through all the answers to see if they are checked, if a checked answer is the correct answer, add 1 to the score
+        for (var j = 0; j < questions[a].answers.length; j++) {
+            // if an answer is checked, then see if it is the correct answer
+            if (questionAnswers[j].checked == true) {
+
+                // if correct answer, add to score
+                if (j == correct) {

```

Comment:
You'd really want to do an if-else here, rather than and if-elseif, but also try to avoid if-elses entirely - Nathan will happily explain.